flutter/lib/resources/result_manager.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:convert';
+import 'dart:io';
+
+class ResultManager {
+  static const _jsonResultFileName = 'result.json';
+  late final String _jsonResultPath;
+
+  ResultManager(String applicationDirectory) {
+    _jsonResultPath = '$applicationDirectory/$_jsonResultFileName';
+  }
+
+  Future<void> write(List<Map<String, dynamic>> content) async {
+    final jsonFile = File(_jsonResultPath);
+
+    final jsonEncoder = JsonEncoder.withIndent('  ');
+    var encoded = jsonEncoder.convert(content);
+    // needed to match android behavior
+    encoded = encoded.replaceAll('/', '\\/');
+    await jsonFile.writeAsString(encoded);
+  }
+
+  Future<String> get json async {
+    final file = File(_jsonResultPath);
+
+    if (await file.exists()) {
+      return file.readAsString();
+    }
+
+    return '';
+  }
```

Comment:
It is meant to be used as `resultManager.json`:
https://github.com/mlcommons/mobile_app_open/blob/214521a9c5098820d7dec5a70bde7273a3beaeae/flutter/lib/ui/result_screen.dart#L361
I think the word `result` would be redundant here.
And I think it's hard to confuse `resultManager.json` as being json representation of the class.

I'm not a big fan of this "getter" syntax, though. Maybe it would be better to make it a function with name like `readJson`.
What do you think?

Reply:
Because of the `Manager` in the class name, it's not clear that `json` means the result.
I think `read()` would be enough. Like other methods `write()`, `delete()`.

Reply:
fixed